app/markdown/long_running_actions.md

```diff
@@ -0,0 +1,172 @@
+
+    const ajax = controller.getServices('ajax');
+
+    module.addServices({
+      atInterval: (url, signals, interval = TEN_SECONDS) => {
+        const {success, error} = signals;
+
+        return window.setInterval(() => {
+          ajax.get(url)
+            .then(response => {
+              controller.getSignals(success)({
+                data: response.data
+              });
+            })
+            .catch(response => {
+              controller.getSignals(error)({response});
+            });
+        }, interval);
+      }
+    });
+  }
+}
+
+```
+`httpPoller` uses the `controller.addServices()` function to expose an `atInterval` method to the `services` argument in your actions. `atInterval` takes a `url` to poll and an object defining `success` and `error` signal names.
+
+Using `window.setInterval` we query our endpoint once every 10 seconds. In the `then()` callback we use the `controller.getSignals()` method to get our signal by name and call it with the returned `message_counts` data. Same goes for our `catch()` callback.
+
+#### Defining the Action
+
+Lets use the `httpPoller` service in our action:
+
+```js
+function pollMessageCounts({state, services, output}) {
+  const {httpPoller} = services;
+  const id = state.get('currentUser.id');
+
+  httpPoller.atInterval(`/api/message_counts/${id}`, {
+    success: 'navbar.messageCountsFetched',
+    error: 'navbar.errorFetchingMessageCounts'
+  })
+}
+```
+
+We grab our `httpPoller` service from the `services` key passed to our action. Then, we pass in *success* and *error* signals to be triggered every 10 seconds.
+
+#### Tying it Together with Signals
+
+Now we can implement our signals and their corresponding action chains in our `Navbar` module and define our component:
+
+```js
+// ./modules/navbar/index.js
+
+export default (options = {}) => {
+  return (module) => {
+
+    module.addSignals({
+      mounted: [
+        pollMessageCounts
+      ],
+      messageCountsFetched: [
+        copy('input:data', 'state:navbar.messageCounts')
+      ],
+      errorFetchingMessageCounts: [
+        // handle error here
+      ]
+    })
+  }
+}
+```
+
+Now we display the `messageCounts` in our component:
+
+```js
+import React,{Component} from 'react';
+import {connect} from 'cerebral-view-react';
+
+export default connect({
+  messageCounts: 'navbar.messageCounts'
+},
+  function NewMessagesIndicator(props) {
+    const newCount = props.messageCounts.new_count
+
+    return (
+      <div>{newCount ? newCount : '--'}</div>
+    )
+  }
+)
+```
+
+And we're done!
+
+#### Summary
+
+- Use Cerebral's *module* interface to expose inputs from the "outside" world as *signals* in your application.
+- Services are passed to *actions* and *actions* can trigger **new** signals from long running, possibly async processes
+- Since all inputs to your app are explicitly listed in your signal definition you can easily understand their behavior and view them in the debugger
```

Comment:
Think it is important to note that these events will possibly happen multiple times. Unlike an ajax request that only has one success or error :)